### PR TITLE
Make single pixel translucent attempt

### LINKED ISSUE #163: Make a single pixel slightly translucent for optimal Twitter image quality
Uploading an image to Twitter decreases the quality. This does not happen for images with transparency.

How about adding an option to save the image, but with a single pixel (e.g. the top-left one) slightly translucent (`0xFE` opacity instead of `0xFF`). @kosamari built a tool for just that: https://github.com/kosamari/translucent But it would be wonderful if this was built-in to Carbon.

### PR DESCRIPTION
**Pull request by mfix22**
@mathiasbynens does this look right? 

@briandennis do you think this is worth having to maintain? Does it get us enough benefit?

- Closes #163

### CODE REVIEW

File: pages/editor.js [hunk 1]
```diff
@@ -111,7 +111,32 @@ class Editor extends React.Component {
...
+        const image = new Image()
+        image.addEventListener(
+          'load',
+          function() {
+            canvas.width = image.width
+            canvas.height = image.height
+            context.drawImage(image, 0, 0, canvas.width, canvas.height)
+
+            const imageData = context.getImageData(0, 0, canvas.width, canvas.height)
+            if (imageData.data[3] >= 255) {
```
Comment (mathiasbynens) [inline pages/editor.js]: The `> 255` case is not possible, so this could just be `== 255`.

File: pages/editor.js [hunk 2]
```diff
@@ -111,7 +111,32 @@ class Editor extends React.Component {
...
+              imageData.data[3] = 254
+            }
+
+            const newUri = canvas.toDataURL('image/png')
+            canvas.remove()
+            resolve(newUri)
+          },
+          false
```
Comment (mathiasbynens) [inline pages/editor.js]: Nit: `false` is the implicit default, so let’s just omit it? `foo.addEventListener(event, fn, false)` → `foo.addEventListener(event, fn)`

File: pages/editor.js [hunk 3]
```diff
@@ -111,7 +111,32 @@ class Editor extends React.Component {
...
 
-    return domtoimage.toPng(node, config)
+    return domtoimage.toPng(node, config).then(uri => {
+      return new Promise(function(resolve, reject) {
+        const canvas = document.createElement('canvas')
+        const context = canvas.getContext('2d')
+        const image = new Image()
+        image.addEventListener(
+          'load',
+          function() {
```
Comment (mathiasbynens) [inline pages/editor.js]: Could be an arrow function to match the rest of the coding style

File: pages/editor.js [hunk 4]
```diff
@@ -111,7 +111,32 @@ class Editor extends React.Component {
...
+            canvas.width = image.width
+            canvas.height = image.height
+            context.drawImage(image, 0, 0, canvas.width, canvas.height)
+
+            const imageData = context.getImageData(0, 0, canvas.width, canvas.height)
+            if (imageData.data[3] >= 255) {
+              imageData.data[3] = 254
+            }
+
+            const newUri = canvas.toDataURL('image/png')
```
Comment (mathiasbynens) [inline pages/editor.js]: maybe `newUri` → `newUrl` to match other variable names in this file

**Comment by mfix22** [comment]
@mathiasbynens thanks so much for the review! Will let @briandennis take a look too before merging

**Comment by mfix22** [comment]
Hey @mathiasbynens turns out someone brought up this issue awhile ago and @jakedex fixed it here: https://github.com/dawnlabs/carbon/pull/104

Have you seen an image not being uploaded as a PNG?

**Comment by mathiasbynens** [comment]
> Have you seen an image not being uploaded as a PNG?

I have not, actually! I should have checked. Apologies 😊

**Comment by jakedex** [comment]
No worries, @mathiasbynens 👍  Closing this for now—

**Pull request closed without merging**